### PR TITLE
fix(ci): Add systemd notification to all mme related services.

### PR DESCRIPTION
**Pull request by crasu**
## Summary

Follow up to #14177. This PR converts all mme related services to systemd notification in order to only mark services as "active" in systemd which have fully started and remove the timeouts in the test.

## Test Plan

Integ run https://github.com/crasu/magma/actions/runs/3314249640 (one flaky test failed due to different reason)

### CODE REVIEW

File: lte/gateway/python/magma/mobilityd/main.py
Comment (wolfseb) [inline lte/gateway/python/magma/mobilityd/main.py]: Just for my understanding: is the reason for only giving mobilityd and pipelined these notifiers because they are the only python services that take some time to restart?

File: lte/gateway/python/magma/mobilityd/BUILD.bazel
Comment (LKreutzer) [inline lte/gateway/python/magma/mobilityd/BUILD.bazel]: Should be `requirement("sdnotify"),`

File: lte/gateway/python/magma/pipelined/BUILD.bazel
Comment (LKreutzer) [inline lte/gateway/python/magma/pipelined/BUILD.bazel]: Should be `requirement("sdnotify"),`

**Review by LKreutzer — CHANGES_REQUESTED** [review]

File: bazel/external/requirements.in
Comment (LKreutzer) [inline bazel/external/requirements.in]: Would it make sense to add a `,<1` to avoid an unintentional major version upgrade?

File: bazel/external/requirements.txt
Comment (LKreutzer) [inline bazel/external/requirements.txt]: Could we move this above line 1627, just to avoid any confusion? This should have been auto-generated there.

File: bazel/external/requirements.txt
Comment (LKreutzer) [inline bazel/external/requirements.txt]: Maybe we could add a 
`    # via -r requirements.in`
in the line below. This should have been auto-generated.

File: lte/gateway/python/integ_tests/s1aptests/test_3485_timer_for_default_bearer_with_mme_restart.py
Comment (LKreutzer) [inline lte/gateway/python/integ_tests/s1aptests/test_3485_timer_for_default_bearer_with_mme_restart.py]: Just out of curiosity, was this change done by the formatter?

**Review by LKreutzer — CHANGES_REQUESTED** [review]
Could you explain why only these services (MME, SCTPD, sessiond, pipelined, mobilityd) are relevant?

File: lte/gateway/python/magma/mobilityd/main.py
Comment (crasu) [reply lte/gateway/python/magma/mobilityd/main.py]: The notifier is added to sctpd and sessiond as well. These services are all restarted together with the mme https://github.com/magma/magma/blob/351a724e32a9cbffed184ba660152fe3290696a3/lte/gateway/deploy/roles/magma/files/systemd_bazel/magma%40mme.service#L13

File: lte/gateway/python/integ_tests/s1aptests/test_3485_timer_for_default_bearer_with_mme_restart.py
Comment (crasu) [reply lte/gateway/python/integ_tests/s1aptests/test_3485_timer_for_default_bearer_with_mme_restart.py]: no just added some debug code at this points and accidential removed these lines.

**Comment by crasu** [comment]
Rebase to latest master after merge of #14177 and resolved review comments.

File: lte/gateway/python/magma/mobilityd/BUILD.bazel
Comment (LKreutzer) [reply lte/gateway/python/magma/mobilityd/BUILD.bazel]: This is still an issue and failing CI.

**Review by LKreutzer — CHANGES_REQUESTED** [review]

File: lte/gateway/c/sctpd/src/sctpd.cpp
Comment (LKreutzer) [reply lte/gateway/c/sctpd/src/sctpd.cpp]: Could we address these cpplint warnings?

File: lte/gateway/c/session_manager/sessiond_main.cpp
Comment (LKreutzer) [reply lte/gateway/c/session_manager/sessiond_main.cpp]: Could we address these cpplint warnings?

**Comment by crasu** [comment]
> Could you explain why only these services (MME, SCTPD, sessiond, pipelined, mobilityd) are relevant?

These are the services related to the mme. Would be nice to have this everywhere but for now we stick to these services.

**Comment by LKreutzer** [comment]
> > Could you explain why only these services (MME, SCTPD, sessiond, pipelined, mobilityd) are relevant?
> 
> These are the services restarted with the mme. Would be nice to have this everywhere but for now we stick to these services.

Hm, I can't quite see this grouping in this graph (based on the service files):
Screenshot from 2022-10-25 15-02-38
See source for the legend: https://excalidraw.com/#room=9b3ea34b6f337ccab568,F2DMzJxlW8yjKZtjQ2djtA

But I'm fine with only doing this group for now.

**Comment by crasu** [comment]
> > > Could you explain why only these services (MME, SCTPD, sessiond, pipelined, mobilityd) are relevant?
> > 
> > 
> > These are the services restarted with the mme. Would be nice to have this everywhere but for now we stick to these services.
> 
> Hm, I can't quite see this grouping in this graph (based on the service files): Screenshot from 2022-10-25 15-02-38 See source for the legend: https://excalidraw.com/#room=9b3ea34b6f337ccab568,F2DMzJxlW8yjKZtjQ2djtA
> 
> But I'm fine with only doing this group for now.



> > > Could you explain why only these services (MME, SCTPD, sessiond, pipelined, mobilityd) are relevant?
> > 
> > 
> > These are the services restarted with the mme. Would be nice to have this everywhere but for now we stick to these services.
> 
> Hm, I can't quite see this grouping in this graph (based on the service files): Screenshot from 2022-10-25 15-02-38 See source for the legend: https://excalidraw.com/#room=9b3ea34b6f337ccab568,F2DMzJxlW8yjKZtjQ2djtA
> 
> But I'm fine with only doing this group for now.

Well its there its all the service that have an incoming arrow from the mme (apart from sctpd). Scptd is kind of special anyway.

**Review by LKreutzer — APPROVED** [review]
Lgtm

Nit: address the cpplint comments

**Review by sebathomas — APPROVED** [review]

**Review by wolfseb — APPROVED** [review]
lgtm

File: lte/gateway/python/setup.py
Comment (nstng) [inline lte/gateway/python/setup.py:136]: Why not use here the same pinning as in the `requierements.in`? -> `sdnotify>=0.3.2,<1`
Or vice versa - I do not see a version `>=1`
```
pip3 install sdnotify==foo
ERROR: Could not find a version that satisfies the requirement sdnotify==foo (from versions: 0.3.0, 0.3.1, 0.3.2)
```

**Review by mpfirrmann — APPROVED** [review]

**Review by ajahl — APPROVED** [review]

**Comment by crasu** [comment]
@pshelar  @uri200 @ardzoht 
We currently have no active developer in the agw-sessiond group. Could you have a look at the sessiond changes for this pr? We are looking for a better solution in the future.

**Review by MoritzThomasHuebner — APPROVED** [review]
Looks good w.r.t. cwf.

**Comment by crasu** [comment]
Remention as an edit does not send out notifications @prabinakpattnaik @uri200 @GANESH-WAVELABS @prabinakpattnaik

**Comment by Neudrino** [comment]
> Hm, I can't quite see this grouping in this graph (based on the service files):
> [graphics omitted]
> See source for the legend: https://excalidraw.com/#room=9b3ea34b6f337ccab568,F2DMzJxlW8yjKZtjQ2djtA

A possibly unrelated comment out of "the off": Alex investigated the relations between the services in more detail recently and created some more dependency graphs for it within https://github.com/magma/magma/issues/14055.
Those can also be [found in the documentation](https://magma.github.io/magma/docs/next/lte/dev_notes#systemd-unit-configuration-files).

**Review by sebathomas — APPROVED** [review]

**Comment by sebathomas** [comment]
@crasu: Can you rebase? With the changes to the codeowners file, the sessiond group should no longer be required?

File: lte/gateway/python/magma/mobilityd/main.py
Comment (pshelar) [inline lte/gateway/python/magma/mobilityd/main.py:20]: lets add the notification to subscriberDB since mobilityD dependency.

File: lte/gateway/python/magma/pipelined/main.py
Comment (pshelar) [inline lte/gateway/python/magma/pipelined/main.py:231]: same as above pipelineD depends on directoryD.

I think it would be safe to add this notification for all python services. I do not see any downside.
Thanks.

**Review by pshelar — DISMISSED** [review]

File: lte/gateway/python/magma/mobilityd/main.py
Comment (crasu) [reply lte/gateway/python/magma/mobilityd/main.py:20]: There does not seem to be a systemd dependency in orc8r/tools/ansible/roles/gateway_services/files/magma.service for mobilityd and directoryd. How is that handled?

File: lte/gateway/python/magma/pipelined/main.py
Comment (crasu) [reply lte/gateway/python/magma/pipelined/main.py:231]: There does not seem to be a systemd dependency in orc8r/tools/ansible/roles/gateway_services/files/magma.service for mobilityd and directoryd. How is that handled?

Comment (pshelar) [reply lte/gateway/python/magma/pipelined/main.py:231]: looks like bug to me.

**Comment by sebathomas** [comment]
I don't see why this PR cannot be merged? Maybe a bug in Github? (It used to contain the sessiond approver group that no longer exists.)

**Comment by voisey** [comment]
@sebathomas It looks to me like it still needs a pipelined approver

**Review by nstng — APPROVED** [review]
pipelined/bazel lgtm - even if there are missing dependencies between pipelined and other services, this change is already an improvement and re-configuring the dependencies can be a follow-up issue.

**Pull request merged**
